### ISSUE TITLE
Phrases in root url (/)

### ISSUE THREAD

**Issue by tombien** [rainlab/translate-plugin#141]
Hello,
I have a site in two languages. I have a home page (with / url). This page includes content file `home.htm` and `home.en.htm`. I put some phrases to this files. And when I go in my browser to `domain.cz/cs` or `domain.cz/en` phrases are always in default (cs) but content file (home.htm / home.en.htm) loads correctly.
When I make two pages with url `/uvod` and `/home` which includes content files and go to `domain.cz/cs/uvod` or `domain.cz/en/home` all phrases are displayed in correct language.
Is there any bug with loading phrases on root url or am I doing something wrong?
Thank you.

**Comment by jan-vince** [comment]
Hi @tombien,
what "phrases" are? Can you send a content of your files and structure?

**Comment by tombien** [comment]
Hello,
this is Page "Home", url is "/"
`{% content 'uvod.htm' %}
<div class="pure-u-1 news">
<h2>{{ 'Aktuálně ze Seabergu'|_ }}</h2>`

In Translate plugin I have default language cs.

I have two Contents `uvod.htm` and `uvod.en.htm`. When I go in my browser to the `example.cz/cs` and `example.cz/en`, the content file loads correctly. But phrase `{{ 'Aktuálně ze Seabergu'|_ }}`  - from Translate modul - is always in default langugage.

This bug is only when url of page is '/'.

**Comment by jan-vince** [comment]
I had a similar problem with URLs translation (#25) - *which you will also have to solve if you want czech URLs*.

This is **probably a bug** - when I return from /en URL to /, locale remains set to "en".

Rainlab.Translate plugin set active locale into Session. But doesn't change it for URLs without locale (when you try "/", locale remains, but when you try "/cs", locale changes).

*This should be configurable somewhere if one wants to store locale in Session (same as there is Force locale in URL checkbox).*

I use this for all Czech CMS pages in their PHP sections to solve the problem:

    function onInit(){

        if($this->page->url == "/"){   // for root page; change to /contact for Contact page ...
            Session::put('rainlab.translate.locale', 'cs');   
        }

    }

----

What I don't undestand is why do you have {{ 'Aktuálně ze Seabergu'|_ }} inside of 'uvod.htm'? 
This file should have Czech texts, doesn't it? Why try to translate it? 

Maybe I am missing something...


I tried the structure like this:

**CMS page home.htm**:

```
title = "Homepage"
url = "/"
is_hidden = 0
==
<?php
function onInit(){
    if($this->page->url == "/"){
        Session::put('rainlab.translate.locale', 'cs');   
    }
}
?>
==
{% content 'locale.htm' %}
```

**CMS content file locale.htm:**

`Český obsah`

**CMS content file locale.en..htm:**

`English content`


For URL "/" I get locale.htm content, for "/en" I get locale.en.htm content.

**Comment by tombien** [comment]
Hello,
I have a Page (CMS > Pages) "Home" and two CMS>Contents (`uvod.htm`, `uvod.en.htm`) - in Page there are only some layouts phrases, like "Aktuálně ze Seabergu" etc. - this is a header for a blog posts, which are included after this header. Also in footer I have some copyrights messages, that I need to translate. In `uvod.htm` and `uvod.en.htm` I have main content of home page.

We solve this, that we delete routing of language prefix in `routes.php` (lines 30-46) in Translate modul. And after this we have `/cs` and `/en` home pages. And one page with `/` and this page is redirected to `/cs`. And other pages have url like this `/cs/kontakty`, `/en/contacts` instead of `/kontakty` and `/contacs`

I still do not try new stable version, maybe it is solved another way.

**Comment by jan-vince** [comment]
I am on stable with Translation plugin v1.1.5. 
*And you fight same problems as I do as I usually make CS/EN/DE websites.*

Translation plugin has two problems right now: no URLs translation and no choice (plural) translations.
 * First can be bypassed with #25 
 * Second with some IFs statements.

Non is perfect, but it works. And October is far the best CMS I have ever used so it is worth the problems :)

*I use Static pages plugin where my customers can translate content themselves - so I don't need separate files (like uvod.en.htm).*

![snimek obrazovky 2016-07-07 v 14 57 28](https://cloud.githubusercontent.com/assets/1104383/16653824/2758a0b0-4453-11e6-8cea-77c1089f573e.png)

----

So I believe **there is a bug in Translation plugin** and we should keep this issue open:

 * Navigating to URL without locale slug (like `/kontakt `or `/` ) doesn't change app locale to default language (it remains the last used locale)

**Comment by tombien** [comment]
Yes, I think, there is a globally problem - page should belong to one specific language. We made our customization as I wrote before. After this, you can say this page is `/cs/kontaky`, `/en/contacs` etc.

Actually works all of this in default set up: `/cs/contacts, /en/kontakty, /contacs, /kontakty` and it is not good.

In new version, you can translate Static page, but you can not translate Url, this is still problem and not good solution.

For me for now is best my Translate plugin customization - switch off language routing, and put this to the Page Url. I have some content in CMS > Pages and some in Static pages (depends on which content I want to make easily editable for customer)

**Comment by daftspunk** [comment]
Fixed by #25
